### DISCUSSION TITLE
Recover from trigger ERROR state

### DISCUSSION THREAD

**Discussion by giacomodeliberali** [quartznet/quartznet#1077] [Q&A]
We are using SQL job store and we run in a situation where, for some deployment issues, some CRON triggers are ended up in `ERROR` state (and thus are not going to be executed in the future). Is there any built-in way to match all triggers in such `ERROR` state to resume them all (so updating to `WAITING`)? Any suggestions?

[Similar SO thread  here.](https://stackoverflow.com/questions/32273540/recover-from-trigger-error-state-after-job-constructor-threw-an-exception)

**Comment by lahma** [comment]
I'd rather understand the root cause than apply bandage for symptom. Which version of Quartz are you running?

**Reply by giacomodeliberali** [reply]
We are on 3.0.7. The root cause was a dll mismatch that prevented jobs to be instanciated by our `JobFactory`

**Reply by lahma** [reply]
Please use the latest version, it has fixes to some state handling too. In your you might need to update database manually, but yeah - if jobs cannot be instantiated it's considered quite bad thing and Quartz will flag them as broken.

**Reply by giacomodeliberali** [reply]
Ok, for *updating the db manually* you mean updating the triggers state? Like `UPDATE QRTZ_TRIGGERS SET [TRIGGER_STATE] = 'WAITING' WHERE [TRIGGER_STATE] = 'ERROR'`?

Thanks for the quick support btw 😃

**Reply by lahma** [reply]
Yes, but try against a backup first that there are no surprises.
